src/controller/c/robot.c

```diff
@@ -1279,26 +1315,30 @@ static char *compute_socket_filename() {
         continue;
       char **r = realloc(filenames, (count + 1) * sizeof(char *));
       if (!r) {
-        fprintf(stderr, "Cannot allocate memory for listing \"%s\" folder.\n", folder);
+        fprintf(stderr, "Cannot allocate memory for listing \"%s\" folder.\n", ipc_folder);
         exit(EXIT_FAILURE);
       }
       filenames = r;
       filenames[count++] = strdup(de->d_name);
     }
     closedir(dr);
-    if (count == 0)
+    if (count == 0) {
+      snprintf(error_buffer, ERROR_BUFFER_SIZE,
+               "The Webots simulation has not yet been started or there is no robot set with an <extern> controller");
```

Comment:
It would be nice here to have a message slightly different from the one at line 1289 (easier to identify where does a problem come from).